### PR TITLE
Do not stop attached environment

### PR DESCRIPTION
**Pull request by kokosing**
Do not stop attached environment

Environment to which `test run` is attaching is controlled by separate
process and so it should be stopped by that process.
Thanks to this environment can be reused multiple times.

### CODE REVIEW

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java [hunk 1]
```diff
@@ -150,6 +157,11 @@ private Environment tryStart()
 
     public void stop()
     {
+        if (attached) {
```
Comment (iirekm) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java:160]: does it relly solve `Could not attach tests to unhealthy environment singlenode-privacera`???

nothing was stopped at my machine, containers were still working as expected, the problem was somewhere in `allCOntainersHealthy` probably

Comment (kokosing) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java:160]: I removed that in https://github.com/prestosql/presto/pull/5415/commits/eed10acea50d5e566c00461aefc5c2082fe2ca7c

**Review by iirekm — APPROVED** [review]

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java [hunk 2]
```diff
@@ -383,10 +395,10 @@ private static void setContainerAutoRemove(CreateContainerCmd createContainerCmd
         private static List<Ulimit> standardUlimits()
         {
             return ImmutableList.of(
-                // Number of open file descriptors
-                new Ulimit("nofile", 65535L, 65535L),
-                // Number of processes
-                new Ulimit("nproc", 8096L, 8096L));
+                    // Number of open file descriptors
+                    new Ulimit("nofile", 65535L, 65535L),
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java:399]: unrelated

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java [hunk 3]
```diff
@@ -150,6 +157,11 @@ private Environment tryStart()
 
     public void stop()
     {
+        if (attached) {
+            // do not stop attached, externally controlled environment
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java:161]: The fact that stop does not stop is not intuitive.

perhaps, make it `throw illegal state` and update  `test attach` not to call stop, if there is nothing to stop.
or, just change the `test attach` flow and then no changes needed in Env clas


also, can the test container leak in `test attach` mode? 
(i hope it's controller by ryuk and "safe to leak")

**Comment by findepi** [comment]
@iirekm cool to see `test attach` being used :)

**Comment by iirekm** [comment]
> @iirekm cool to see `test attach` being used :)

it doesn't work at all for my case

**Comment by kokosing** [comment]
>> @iirekm cool to see test attach being used :)

> it doesn't work at all for my case

I love you both ❤️

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java (hunk 3 above)
Comment (kokosing) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java:161]: > The fact that stop does not stop is not intuitive.

Fact. I will move this `if` to `close` method and throw in  `stop`

> also, can the test container leak in test attach mode?

I tested it and it was cleared. I will test it more with interruptions.

**Comment by kokosing** [comment]
@findepi AC

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/EnvironmentUp.java [hunk 4]
```diff
@@ -144,7 +144,7 @@ public Integer call()
                 builder.configureContainers(Standard::enablePrestoJavaDebugger);
             }
 
-            Environment environment = builder.build(getStandardListeners(environmentLogPath));
+            Environment environment = builder.build(getStandardListeners(environmentLogPath), false);
```
Comment (wendigo) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/EnvironmentUp.java:147]: can you add .setAttached() method to Builder instead?

**Review by findepi — APPROVED** [review]
but mind @wendigo 's

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java (hunk 3 above)
Comment (kokosing) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java]: > I tested it and it was cleared. I will test it more with interruptions.

It works as expected.

**Review by kokosing — COMMENTED** [review]
@wendigo AC

**Review by wendigo — APPROVED** [review]
